pkg/plugins/models.go

```diff
@@ -280,6 +291,11 @@ func (e Error) AsErrorCode() ErrorCode {
 	return ""
 }
 
+func (e Error) With(err error) Error {
```

Comment:
This one confuses me a bit. In places where I already specify an error code, it's not clear if passing an error will do some magic to overwrite that for example.

 Could this be this instead maybe?
```suggestion
func (e Error) WithMessage(message string) Error {
```

Reply:
sure :+1: